### PR TITLE
Retrieve original value from a point in a MultidimensionalSparseTuningProblem

### PR DESCRIPTION
**Pull request by DavidPoliakoff**
This allows a user to get a configuration from a MSTP given a point in its configuration space. This means that if a tool can produce code representing its heuristics, the generated code can be used in a library to configure itself _without_ a tuner

### CODE REVIEW

**Comment by Rombur** [comment]
You need to reformat the code.

**Review by Rombur — APPROVED** [review]
This looks good but I think it would be useful to have a test to make sure it works as intended

**Comment by DavidPoliakoff** [comment]
@Rombur added tests. Unfortunately, even with clang-format from clang 8.0.0, clang-format -i on the failing files isn't changing anything for me (and it does pick up the .clang-format we want). I may just manually change what the patch says to do, but it's a bit obnoxious to have to fight this hard over " &" versus "& "

**Comment by Rombur** [comment]
I think there is a problem with the last commit. The commit message says that you added a test but the commit itself just removes a blank line.

**Comment by masterleinad** [comment]
Also, there a lot of whitespace changes that are out-of-sope... Not sure what's going on.

**Comment by DavidPoliakoff** [comment]
Again, if it passes CI, I don't worry further about formatting. It's just _not_ the important thing to me

**Comment by masterleinad** [comment]
I think the still you wanted to add is still missing.

File: core/unit_test/tools/TestBuiltinTuners.cpp [hunk 1]
```diff
@@ -93,6 +93,10 @@ int main(int argc, char* argv[]) {
     end_context(kernel_context);
     begin_context(kernel_context);
     set_input_values(kernel_context, 1, &md_kernel_value);
+    auto begin_point = new_team_tuner.get_point(0.0, 0.0, 0.0);
+    assert(std::get<0>(begin_point) == 1);
+    auto end_point = new_team_tuner.get_point(0.9, 0.0, 0.0);
+    assert(std::get<0>(end_point) == 5);
```
Comment (masterleinad) [inline core/unit_test/tools/TestBuiltinTuners.cpp:99]: Would you mind explaining why we expect the returned value to be 1 and 5 respectively?

Comment (DavidPoliakoff) [reply core/unit_test/tools/TestBuiltinTuners.cpp:99]: 0.0 of the way through the outermost dimension of this tuner is the first option (1). 0.9 of the way through the outermost dimension is 5

Comment (DavidPoliakoff) [reply core/unit_test/tools/TestBuiltinTuners.cpp]: cfe8be9

File: core/unit_test/tools/TestBuiltinTuners.cpp [hunk 2]
```diff
@@ -93,9 +93,18 @@ int main(int argc, char* argv[]) {
     end_context(kernel_context);
     begin_context(kernel_context);
     set_input_values(kernel_context, 1, &md_kernel_value);
+
+    /**
+     * Note that 0.0 is basically a floating point index into
+     * the outermost index in this, which is the options vector
+     * above. The At 0.0, this will be the first element (1).
+     * At 0.,9 this will be the last element (5)
```
Comment (masterleinad) [inline core/unit_test/tools/TestBuiltinTuners.cpp:101]:
```suggestion
     * At 0.9 this will be the last element (5)
```

**Review by masterleinad — APPROVED** [review]
Otherwise, this looks OK to me.

**Pull request merged**